packages/api/src/auth/events/user-registration-completed.event.ts

```diff
@@ -0,0 +1,5 @@
+import type { UserRegistration } from '@prisma/client';
+
+export class UserRegistrationCompletedEvent {
+  constructor(public readonly payload: UserRegistration) {}
```

Comment:
Database entities should be internal to the module (you can change how you persist data whatever you want, but it should not impact other parts of an app). Here we publish database internals in events to whole application.

Reply:
Isn't `UserRegistration` directly related to the module that sends this event? I mean this is an event that sends all information received by `UserRegistration`. In the current implementation if I for example add a `birthdate` field to registration form I wouldn't need to change it in many places

Reply:
You gave a specific example :) Adding filed is a good example of non-breaking operation. But assume that you remove field from database / split one into two, etc. What with modules which are subscribing for the event? The change will not be limited to just database and publisher, but to event and all subscribers. 
For me, good practice is to define module api (which are events part of - and in this architecture, service methods also) separately and treat database entities as implementation details. 
Another reason: i think it's really unexpected to change db access technology (like prisma to typeorm), but it sometimes happens, if yes - then this will have impact for almost all part of the app, so it's nice to do not leave module boundaries with internals like db access.